src/main/java/de/zalando/aruha/nakadi/config/NakadiConfig.java

```diff
@@ -86,18 +79,36 @@ public ZooKeeperHolder zooKeeperHolder() {
     }
 
     @Bean
+    @Profile("kafka")
     public KafkaLocationManager getKafkaLocationManager() {
         return new KafkaLocationManager();
     }
 
     @Bean
+    @Profile("kafka")
     public KafkaFactory kafkaFactory() {
         return new KafkaFactory(getKafkaLocationManager());
     }
 
     @Bean
-    public KafkaRepositorySettings topicRepositorySettings() {
+    @Profile("kafka")
+    public KafkaRepositorySettings kafkaRepositorySettings() {
         return new KafkaRepositorySettings();
     }
 
+    @Bean
+    public TopicRepository topicRepository() {
+        if (environment.acceptsProfiles("kafka")) {
```

Comment:
Can't agree with you at this point.

a) If we want one Nakadi instance to be able to use different brokers types we will then have to completely rework current architecture, as it will not be possible to achieve with current DI we have.
b) "forces us to specify this profile" sounds very scary but in fact this is just one property in application.yml and that's it.

Also, using spring-profiles opens a theoretical possibility to support other brokers (yes, only one per instance), and I believe this is better than nothing.